### PR TITLE
DOC: Tell people to use only-binary option

### LINKED ISSUE #21711: TST: Nightly wheel asking for setuptools version that Actions does not have
https://github.com/numpy/numpy/blob/368baf45d023084798231e5abd0135783a2fc7c6/pyproject.toml#L5

resulted in such errors in downstream jobs that grabs nightly numpy wheel

```
Ignoring packaging: markers 'platform_machine == "arm64"' don't match your environment
ERROR: Could not find a version that satisfies the requirement setuptools==59.2.0 (from versions: none)
ERROR: No matching distribution found for setuptools==59.2.0
```

https://github.com/astropy/astropy/runs/6818500953?check_suite_focus=true (GitHub Actions ubuntu-latest, Python 3.8)

This setup has been working for a while but suddenly stopped working. I cannot tell if something changed in Actions or in the nightly wheel.

### PR DESCRIPTION
**Pull request by pllim**
Tell people to use only-binary option when installing nightly wheel for downstream testing.

Close #21711

### CODE REVIEW

**Comment by pllim** [comment]
Whoops... I should probably fix the commit message.

File: doc/source/user/depending_on_numpy.rst [hunk 1]
```diff
@@ -37,7 +37,10 @@ Testing against the NumPy main branch or pre-releases
 For large, actively maintained packages that depend on NumPy, we recommend
 testing against the development version of NumPy in CI. To make this easy,
 nightly builds are provided as wheels at
-https://anaconda.org/scipy-wheels-nightly/.
+https://anaconda.org/scipy-wheels-nightly/. Example install command::
+
+    pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+
 This helps detect regressions in NumPy that need fixing before the next NumPy
 release.  Furthermore, we recommend to raise errors on warnings in CI for this
```
Comment (mattip) [inline doc/source/user/depending_on_numpy.rst:45]: This recently came up elsewhere, does it make sense to also recommend a way to turn warnings into errors?
```suggestion
release.  Furthermore, we recommend to use `python -Werrors` to turn
warnings into errors in CI for this
```

Comment (pllim) [reply doc/source/user/depending_on_numpy.rst:45]: We use the pytest "filterwarnings" settings for more control (https://docs.pytest.org/en/stable/how-to/capture-warnings.html). `-Werrors` might also turn unrelated warnings into errors, so YMMV. But if you want that anyway, I can apply your suggestion, so please let me know. Thanks!

**Comment by mattip** [comment]
Thanks @pllim

**Pull request merged**